### ISSUE TITLE
LND sync scanning progress

### ISSUE THREAD

**Issue by frennkie** [raspiblitz/raspiblitz#1366]
- [ ] change to lnd `getinfo`

On LND 0.10.X (?!) the log format seem to have changed. The [00infoBlitz.sh (INFO)](https://github.com/rootzoll/raspiblitz/blob/c6a98161468de1faa717d45f77c9470af181858c/home.admin/00infoBlitz.sh#L271) uses `(height` but the line (with **loglevel DEBUG**)  is:

```
2020-07-20 14:56:39.287 [DBG] LNWL: Filtering block 605826 (00000000000000000000f383c11667be88942faced981f45e42b4a366b5c31a3) with 3159 transactions
2020-07-20 14:56:42.683 [DBG] LNWL: Filtering block 605827 (00000000000000000004ee6591455c105127659213990e6aaa1436a022d5d0de) with 2905 transactions
```

Scanning X/Z <- the X is not changing!

![2020-07-20 15_49_54-Window](https://user-images.githubusercontent.com/6499251/87946431-476c1600-caa2-11ea-9b1b-333071c46e07.png)

```
2020-07-20 14:43:18.283 [INF] LTND: Version: 0.10.4-beta commit=v0.10.4-beta, build=production, logging=default
2020-07-20 14:43:18.283 [INF] LTND: Active chain: Bitcoin (network=mainnet)
2020-07-20 14:43:18.284 [INF] LTND: Opening the main database, this might take a few minutes...
2020-07-20 14:46:03.187 [INF] CHDB: Checking for schema update: latest_version=13, db_version=13
2020-07-20 14:46:03.188 [INF] LTND: Database now open (time_to_open=2m44.903992985s)!
2020-07-20 14:46:03.240 [INF] RPCS: password RPC server listening on 127.0.0.1:20009
2020-07-20 14:46:03.240 [INF] RPCS: password gRPC proxy started at 127.0.0.1:8080
2020-07-20 14:46:03.240 [INF] RPCS: password RPC server listening on 127.0.0.1:10009
2020-07-20 14:46:03.262 [INF] LTND: Waiting for wallet encryption password. Use `lncli create` to create a wallet, `lncli unlock` to unlock an existing wallet, or `lncli changepassword` to change the password of an existing wallet and unlock it.
2020-07-20 14:46:12.358 [INF] LNWL: Opened wallet
2020-07-20 14:46:12.870 [INF] LTND: Primary chain is set to: bitcoin
2020-07-20 14:46:12.977 [INF] LTND: Initializing bitcoind backed fee estimator in CONSERVATIVE mode
2020-07-20 14:46:12.978 [INF] LNWL: Started listening for bitcoind transaction notifications via ZMQ on 127.0.0.1:28333
2020-07-20 14:46:12.977 [INF] LNWL: Started listening for bitcoind block notifications via ZMQ on 127.0.0.1:28332
2020-07-20 14:46:18.850 [INF] LNWL: The wallet has been unlocked without a time limit
2020-07-20 14:46:19.060 [INF] LTND: LightningWallet opened
2020-07-20 14:46:19.220 [INF] SRVR: Proxying all network traffic via Tor (stream_isolation=true)! NOTE: Ensure the backend node is proxying over Tor as well
2020-07-20 14:46:19.490 [INF] HSWC: Restoring in-memory circuit state from disk
2020-07-20 14:46:19.534 [INF] HSWC: Payment circuits loaded: num_pending=0, num_open=0
2020-07-20 14:46:19.605 [INF] HSWC: Trimming open circuits for chan_id=592003:77:1, start_htlc_id=0
2020-07-20 14:46:19.850 [INF] LTND: Channel backup proxy channel notifier starting
2020-07-20 14:46:19.864 [INF] ATPL: Instantiating autopilot with active=false, max_channels=5, allocation=0.600000, min_chan_size=20000, max_chan_size=16777215, private=false, min_confs=1, conf_target=3
2020-07-20 14:46:19.901 [INF] RPCS: RPC server listening on 127.0.0.1:10009
2020-07-20 14:46:19.901 [INF] RPCS: RPC server listening on 127.0.0.1:20009
2020-07-20 14:46:19.942 [INF] RPCS: gRPC proxy started at 127.0.0.1:8080
2020-07-20 14:46:19.945 [INF] LTND: Waiting for chain backend to finish sync, start_height=640028
2020-07-20 14:46:20.401 [INF] LNWL: Started rescan from block 0000000000000000000bafad447afff4d430f97d68f66842ea8c42c089c8c75e (height 605595) for 8 addresses
2020-07-20 14:46:29.434 [INF] RPCS: [debuglevel] changing debug level to: debug
2020-07-20 14:46:31.391 [DBG] LNWL: Filtering block 605599 (0000000000000000000ba919ba32d481703927947745e49126291dcf56d345f6) with 2522 transactions
2020-07-20 14:46:34.839 [DBG] LNWL: Filtering block 605600 (000000000000000000127bef398960e2edb5a1e276267def3d6f3c217191aaf9) with 2600 transactions
2020-07-20 14:46:37.921 [DBG] LNWL: Filtering block 605601 (0000000000000000000055b04770cdc9ca77d73886ba2ddf880b8ee827515d12) with 2151 transactions
2020-07-20 14:46:40.448 [DBG] LNWL: Filtering block 605602 (0000000000000000000fc284cd7da224890e5e4ca00c6479ece8f643f2b7ae0c) with 2980 transactions
2020-07-20 14:46:43.958 [DBG] LNWL: Filtering block 605603 (000000000000000000132b08ab9c361abfca21684ec13891684db263dba49f55) with 3007 transactions
2020-07-20 14:46:47.324 [DBG] LNWL: Filtering block 605604 (000000000000000000112208b4c87fa12b83a6a5ccf3ba6e6a28960767c9af67) with 2309 transactions
2020-07-20 14:46:50.215 [DBG] LNWL: Filtering block 605605 (00000000000000000001a2d9bd2c8de99dabbf80368b5c0c02e5693b29de2b24) with 2586 transactions
2020-07-20 14:46:53.400 [DBG] LNWL: Filtering block 605606 (00000000000000000012b939ae969aeae55a209f33ba3643c9f473e415ee4d84) with 3080 transactions
2020-07-20 14:46:56.963 [DBG] LNWL: Filtering block 605607 (000000000000000000117af6ebe82f0b3d9f708d4d48512681b4db36e1cbc1d7) with 1620 transactions
2020-07-20 14:46:59.382 [DBG] LNWL: Filtering block 605608 (0000000000000000000a82bb01dac73c5efc3204bb12ec5f594458e6c0d967d6) with 2898 transactions
2020-07-20 14:47:02.773 [DBG] LNWL: Filtering block 605609 (000000000000000000095358d008281b79a7d335c6a7c4bae9b75aa8d43ab50b) with 2221 transactions
```

**Comment by frennkie** [comment]
I used this to enble debug logging: `lncli debuglevel --level debug`

**Comment by frennkie** [comment]
The LCD (`00infoLCD.sh`) calls `80scanLND.sh` which uses `/home/admin/config.scripts/blitz.statusscan.sh`. There `getinfo` is called and the `best_header_timestamp` value is used to determine the progress. 

I guess it makes sense to switch the `00infoBlitz.sh` file to use the same way.

**Comment by rootzoll** [comment]
- [ ] change to lnd `getinfo`

**Comment by openoms** [comment]
looking good on the latest master (updated scripts, not with the SD yet - hence the old bitcoin version)
![image](https://user-images.githubusercontent.com/43343391/87993385-41962500-cae2-11ea-9309-1eddd2aab769.png)

![image](https://user-images.githubusercontent.com/43343391/87993531-a8b3d980-cae2-11ea-80f1-8d0618d3ff1a.png)

**Comment by rootzoll** [comment]
Working with the quick fix for v1.6 ... will go with better implementation for v1.7.

**Comment by bitsam** [comment]
@frennkie & @rootzoll  is there a possibility to apply a quick fix to 1.5 release? Like @openoms suggested with an updated script. I would really like to fix this issue on my node running 1.5 release (and manually updated LND to 10.3 & bitcoin to v0.20.0).

**Comment by rootzoll** [comment]
@bitsam you can try to apply the changes frennkie made for the v1.6 release manually to your script and reboot:
https://github.com/rootzoll/raspiblitz/pull/1370/files

**Comment by frennkie** [comment]
@bitsam note that (as stated) #1370 requires LND to be run with debug log level.

**Comment by bitsam** [comment]
@frennkie & @rootzoll , i am running v1.6-2020-08-04.img.gz and notice that the LND sync drops out several times a day, similar to the itial reported error by frennkie. Is the fix a work in progress due to changes with bitcoin v0.20.0 & LND v0.11 ? Or could this be my Rpi3 and SATA HDD are not quick enough to keep up with the LND chain ? What info should I be looking at to see why my node is losing LND sync so often ?

*** LAST 30 LND INFO LOGS ***
sudo tail -n 30 /mnt/hdd/lnd/logs/bitcoin/mainnet/lnd.log
2020-08-12 10:26:18.291 [INF] HSWC: ChannelLink(588731:328:1): starting
2020-08-12 10:26:18.291 [INF] HSWC: Trimming open circuits for chan_id=588731:328:1, start_htlc_id=33
2020-08-12 10:26:18.291 [INF] HSWC: Adding live link chan_id=7bfc7b870a332fab6432ab93d9fe45e1ef92c64d4e899a71315c4fac6ee97103, short_chan_id=588731:328:1
2020-08-12 10:26:18.291 [INF] CNCT: Attempting to update ContractSignals for ChannelPoint(0271e96eac4f5c31719a894e4dc692efe145fed993ab3264ab2f330a877bfc7b:1)
2020-08-12 10:26:18.291 [INF] HSWC: ChannelLink(588731:328:1): HTLC manager started, bandwidth=2414413332 mSAT
2020-08-12 10:26:18.291 [INF] PEER: Negotiated chan series queries with 03a503d8e30f2ff407096d235b5db63b4fcf3f89a653acb6f43d3fc492a7674019
2020-08-12 10:26:18.292 [INF] HSWC: ChannelLink(588731:328:1): attempting to re-resynchronize
2020-08-12 10:26:18.292 [INF] DISC: Creating new GossipSyncer for peer=03a503d8e30f2ff407096d235b5db63b4fcf3f89a653acb6f43d3fc492a7674019
2020-08-12 10:26:18.301 [INF] HSWC: ChannelLink(588731:328:1): received re-establishment message from remote side
2020-08-12 10:26:18.490 [INF] PEER: NodeKey(0247719ca61fa34ce383634a30b9cde6a2b396a8a6e2974dbbd4ebbe93e093ad2c) loading ChannelPoint(9a260a710a92362af6b8f09baee8a94fdfc6bb6fcfb00fa04f4de153d9cb5dd1:0)
2020-08-12 10:26:18.491 [INF] HSWC: Removing channel link with ChannelID(d15dcbd953e14d4fa00fb0cf6fbbc6df4fa9e8ae9bf0b8f62a36920a710a269a)
2020-08-12 10:26:18.491 [INF] HSWC: ChannelLink(628774:976:0): starting
2020-08-12 10:26:18.491 [INF] HSWC: Trimming open circuits for chan_id=628774:976:0, start_htlc_id=0
2020-08-12 10:26:18.491 [INF] HSWC: Adding live link chan_id=d15dcbd953e14d4fa00fb0cf6fbbc6df4fa9e8ae9bf0b8f62a36920a710a269a, short_chan_id=628774:976:0
2020-08-12 10:26:18.492 [INF] CNCT: Attempting to update ContractSignals for ChannelPoint(9a260a710a92362af6b8f09baee8a94fdfc6bb6fcfb00fa04f4de153d9cb5dd1:0)
2020-08-12 10:26:18.492 [INF] PEER: Negotiated chan series queries with 0247719ca61fa34ce383634a30b9cde6a2b396a8a6e2974dbbd4ebbe93e093ad2c
2020-08-12 10:26:18.492 [INF] HSWC: ChannelLink(628774:976:0): HTLC manager started, bandwidth=0 mSAT
2020-08-12 10:26:18.493 [INF] HSWC: ChannelLink(628774:976:0): attempting to re-resynchronize
2020-08-12 10:26:18.492 [INF] DISC: Creating new GossipSyncer for peer=0247719ca61fa34ce383634a30b9cde6a2b396a8a6e2974dbbd4ebbe93e093ad2c
2020-08-12 10:26:18.495 [INF] HSWC: ChannelLink(628774:976:0): received re-establishment message from remote side
2020-08-12 10:26:31.553 [ERR] BTCN: Can't accept connection: unable to accept connection from 104.131.26.124:42919: EOF
2020-08-12 10:27:39.589 [INF] DISC: GossipSyncer(02729134d5ef3297717a8e53539442ba4d08da400c7381d0e9211f0b66c1e46601): no more chans to query
2020-08-12 10:27:39.589 [INF] DISC: GossipSyncer(0217890e3aad8d35bc054f43acc00084b25229ecff0ab68debd82883ad65ee8266): applying gossipFilter(start=2020-08-12 10:27:39.589369328 +1000 AEST m=+1744.312801644, end=2156-09-18 17:55:54.589369328 +1100 AEDT m=+4294969039.312801644)
2020-08-12 10:27:39.589 [INF] DISC: GossipSyncer(03094b63a5ece2c363ab859952017f95bec7bd9e6975f0c8152f9eff83d0b045c2): applying gossipFilter(start=2020-08-12 10:27:39.589692709 +1000 AEST m=+1744.313125442, end=2156-09-18 17:55:54.589692709 +1100 AEDT m=+4294969039.313125442)
2020-08-12 10:27:39.589 [INF] DISC: GossipSyncer(03a503d8e30f2ff407096d235b5db63b4fcf3f89a653acb6f43d3fc492a7674019): applying gossipFilter(start=2020-08-12 10:27:39.589903749 +1000 AEST m=+1744.313336013, end=2156-09-18 17:55:54.589903749 +1100 AEDT m=+4294969039.313336013)
2020-08-12 10:27:40.290 [INF] CRTR: Pruning channel graph using block 00000000000000000009452bc324000526ae9bca2dc501669477dcf8571b9670 (height=643294)
2020-08-12 10:27:41.122 [INF] CRTR: Block 00000000000000000009452bc324000526ae9bca2dc501669477dcf8571b9670 (height=643294) closed 0 channels
2020-08-12 10:27:42.842 [INF] NTFN: New block: height=643294, sha=00000000000000000009452bc324000526ae9bca2dc501669477dcf8571b9670
2020-08-12 10:27:42.951 [INF] UTXN: Attempting to graduate height=643294: num_kids=0, num_babies=0
2020-08-12 10:28:06.137 [ERR] BTCN: Can't accept connection: unable to accept connection from 180.150.78.6:47632: EOF

**Comment by rootzoll** [comment]
I removed the "scanning progress" completly on the ´00infoBlitz.sh´ .. it will just show "Waiting for Chain Sync" now when LND signals that `synced_to_chain` is false from `getinfo`. Normally the other blue screen would jump in here anyway and that uses the better `blitz.statusscan.sh` information/progress.

Also I changed the color from red to amber ... because its not a super critical event ... its even normal that LND might take a second to scan a block and if the raspiblitz measures at exactly that point in time the info on the LCD might stay longer on the screen (until next refresh) than it even takes fro LND to catch up. Also I dont think your LND is "offline" during that time ... to forward or receive it should not be crucial to have the latest block in that second. You just need to know if channel force closed ... and if that channel is making in that second a payment you would have a newer closing transaction to punish that channel partner.
